warehouse/admin/views/users.py

```diff
@@ -593,3 +598,31 @@ def user_recover_account_complete(user: User, request):
         queue="success",
     )
     return HTTPSeeOther(request.route_path("admin.user.detail", username=user.username))
+
+
+@view_config(
+    route_name="admin.user.burn_recovery_codes",
+    require_methods=["POST"],
+    permission=Permissions.AdminUsersWrite,
+    uses_session=True,
+    require_csrf=True,
+    context=User,
+)
+def user_burn_recovery_codes(user, request):
+    codes = request.POST.get("to_burn", "").strip().split()
+    if not codes:
+        request.session.flash("No recovery codes provided", queue="error")
+
+    else:
+        user_service = request.find_service(IUserService, context=None)
+        n_burned = 0
+
+        for code in codes:
+            try:
+                if user_service.check_recovery_code(user.id, code):
```

Comment:
Will this add more events to a user's security history, one per code burned?

Reply:
No, the events are generated outside of the `check_recovery_code` function when the user actually burns a code.